examples/model_selection/plot_train_error_vs_test_error.py

```diff
@@ -15,24 +15,27 @@
 # Author: Alexandre Gramfort <alexandre.gramfort@inria.fr>
 # License: BSD 3 clause
 
+# %%
+# Generate sample data
+# --------------------
 import numpy as np
 from sklearn import linear_model
 
-# #############################################################################
-# Generate sample data
 n_samples_train, n_samples_test, n_features = 75, 150, 500
 np.random.seed(0)
 coef = np.random.randn(n_features)
 coef[50:] = 0.0  # only the top 10 features are impacting the model
 X = np.random.randn(n_samples_train + n_samples_test, n_features)
 y = np.dot(X, coef)
 
+# %%
```

Comment:
I find the code above quite complex. You can replace it with our available helper function from scikit-learn:

```python

from sklearn.datasets import make_regression
from sklearn.model_selection import train_test_split

n_samples_train, n_samples_test, n_features = 75, 150, 500
X, y, coef = make_regression(
    n_samples=n_samples_train + n_samples_test,
    n_features=n_features,
    n_informative=50,
    shuffle=False,
    noise=1.0,
    coef=True,
)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, train_size=n_samples_train, test_size=n_samples_test, shuffle=False
)
```